ambry-account/src/main/java/com/github/ambry/account/MySqlAccountService.java

```diff
@@ -104,9 +108,33 @@ protected boolean removeEldestEntry(Map.Entry<String, Boolean> eldest) {
             return size() > cacheMaxLimit;
           }
         }));
-    this.writeCacheAfterUpdate = config.writeCacheAfterUpdate;
+
+    accountMigrationService =
+        new AccountMigrationService(backupFileManager, mySqlAccountStore, needRefresh, lastSyncTime,
+            recentNotFoundContainersCache, accountInfoMapRef, accountServiceMetrics.getAccountServiceMetrics(),
+            infoMapLock, config, notifier, scheduler);
+
+    if (config.enableNewDatabaseForMigration) {
+      backupFileManagerNew =
+          new BackupFileManager(accountServiceMetrics, config.backupDirNew, config.maxBackupFileCount);
+      recentNotFoundContainersCacheNew = Collections.newSetFromMap(
+          Collections.synchronizedMap(new LinkedHashMap<String, Boolean>(cacheInitialCapacity, cacheLoadFactor, true) {
+            protected boolean removeEldestEntry(Map.Entry<String, Boolean> eldest) {
+              return size() > cacheMaxLimit;
+            }
+          }));
+      accountInfoMapRefNew = new AtomicReference<>(new AccountInfoMap(accountServiceMetrics.getAccountServiceMetricsNew()));
```

Comment:
we don't need to create these objects in MysqlAccountService, we should create these objects in AccountMigrationSerivce.

Reply:
After discussion, using the supplier for lazy connection.